### DISCUSSION TITLE
Что такое middleware и как с ним работать

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#61] [ASP.NET Core]
Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:
https://github.com/cleannetcode/LessonMonitor/tree/homework/aspnet%2F3

По выполнению работы, вам необходимо создать пулл реквест из вашей ветки с выполненной домашкой в основной репозиторий.

Почитать:
https://metanit.com/sharp/aspnet5/2.2.php
https://metanit.com/sharp/aspnet5/2.3.php
https://metanit.com/sharp/aspnet5/2.22.php
https://metanit.com/sharp/aspnet5/2.4.php
https://metanit.com/sharp/aspnet5/2.18.php
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-5.0

Сделать:
Создать свой компонент middleware как отдельный класс, который будет сохранять запросы в файл. 
Создать свой компонент middleware как метод Use, который будет сохранять запросы в файл. 
Получится своего рода файл лог.

Проверить:
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by eniluck** [comment]
Вот тут незакончено предложение ссылкой "Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:".
Мы продолжаем предыдущую домашку например от ветки homework/aspnet/2 делаем ветку homework/aspnet/3 ?

**Reply by pingvin1308** [reply]
нужно колоть новую ветку от https://github.com/cleannetcode/LessonMonitor/tree/homework/aspnet%2F3

**Reply by eniluck** [reply]
ок. добавил remote repo - cleannetcode. сделал checkout ветки homework/aspnet/3 . Установил upstream на origin. Сделал push.
